src/ucp/wireup/select.c

```diff
@@ -590,8 +590,8 @@ ucp_wireup_tl_iface_latency(ucp_context_h context,
 
     if (remote_iface_attr->addr_version == UCP_OBJECT_VERSION_V1) {
         /* Address v1 contains just latency overhead */
-        return ucp_tl_iface_latency(context, &iface_attr->latency) +
-               remote_iface_attr->lat_ovh;
+        return (iface_attr->latency.c + remote_iface_attr->lat_ovh) / 2 +
```

Comment:
IMO it should be average of local and remote:
`(ucp_tl_iface_latency(context, &iface_attr->latency) + remote_iface_attr->lat_ovh) / 2 `

Reply:
also, probably need to update line 600 (address version 2) as well

Reply:
add ( ) around / expr